### PR TITLE
Add favicon to doc

### PR DESCRIPTION
**Pull request by harupy**
This PR adds a favicon to the doc. I downloaded the favicon from [the optuna official website](https://optuna.org/).

It looks like this:

Screen Shot 2019-12-16 at 1 15 07

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -93,6 +93,12 @@
 #
 # html_theme_options = {}
 
+# The name of an image file (relative to this directory) to use as a favicon of
+# the docs.  This file should be a Windows icon file (.ico) being 16x16 or 32x32
+# pixels large.
+#
```
Comment (hvy) [inline docs/source/conf.py]: A nit but let's remove the comment as it basically duplicates the documentation.

Comment (harupy) [reply docs/source/conf.py]: ok I'll remove it.

Comment (harupy) [reply docs/source/conf.py]: @hvy Removed it.

**Comment by hvy** [comment]
Did you get the favicon to render on other browsers besides Chrome? It doesn't seem to work with Safari/Edge.

File: docs/source/conf.py (hunk 1 above)
Comment (hvy) [reply docs/source/conf.py]: Thanks for the swift action!

**Comment by harupy** [comment]
No, I didn't. I tested only on Chrome.

**Comment by harupy** [comment]
I'm testing on Safari.

Left: with [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) (seems to be working.). 
Right: Open file with Safari (doesn't work).

Screen Shot 2019-12-16 at 12 53 51

**Pull request merged**